### PR TITLE
[Feature] v3 Integration

### PR DESCRIPTION
**Pull request by fonstack**
## Summary by CodeRabbit

- **New Features**
  - Added support for `HATSVaultV3` and related functionalities in vault systems.
  - Introduced a new version `v3` for vaults, enhancing management and payouts.
  - Added new components and updated logic in the web interface to support `v3` vaults.

- **Enhancements**
  - Improved vault editing tools and payout calculations for different vault versions.
  - Enhanced user interface elements to adapt to new vault and payout structures.

- **Bug Fixes**
  - Adjusted payout allocation logic and fixed issues with multi-payout calculations.

- **Documentation**
  - Updated terms and labels in the user interface to reflect new features and changes.

### CODE REVIEW

File: packages/shared/src/config/chains.ts [hunk 1]
```diff
@@ -53,23 +55,26 @@ export const ChainsConfig: { [index: number]: IChainConfiguration } = {
...
+  //   chain: wagmiChains.goerli,
+  //   subgraph: "https://api.thegraph.com/subgraphs/name/hats-finance/hats_goerli",
+  //   coingeckoId: undefined,
+  //   govMultisig: "0xFc9F1d127f8047B0F41e9eAC2Adc2e5279C568B7",
+  //   whitelistedReviewers: ["0x5b7B59862447Cf3dE479C8166cc2784cbf8e53D6"],
+  //   uniswapSubgraph: undefined,
+  //   paymentSplitterFactory: "0x728654Bb8E69b9978E79657332a0843606d64FF4",
+  //   infuraKey: "goerli",
+  //   provider: "https://eth-goerli.g.alchemy.com/v2/HMtXCk0FyIfbiNAVm4Xcgr8Eqlc5_DKd",
+  // },
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply packages/shared/src/config/chains.ts:74]: please remove

File: packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/SplitPayoutAllocation/components/AdditionalBeneficiariesInfo.tsx [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+  const hatsGovFee = +vault?.governanceHatRewardSplit / 100 / 100;
+  const governancePercentage = totalToPay * hatsGovFee;
+  const hackersPercentage = totalToPay * (1 - hatsGovFee);
+  const depositorsPercentage = 100 - totalToPay;
+
+  const hackersPoints = beneficiaries.reduce((acc, beneficiary) => acc + +beneficiary.percentageOfPayout, 0);
+  const governancePoints = (governancePercentage * hackersPoints) / hackersPercentage;
+  const depositorsPoints = (depositorsPercentage * hackersPoints) / hackersPercentage;
+
+  if (isNaN(depositorsPercentage) && depositorsPercentage > 0) return null;
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/SplitPayoutAllocation/components/AdditionalBeneficiariesInfo.tsx:39]: i think the condition is wrong here

File: packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/usePayoutAllocation.ts [hunk 3]
```diff
@@ -46,13 +46,22 @@ export const usePayoutAllocation = (
...
   // We need to multiply the results by the percentage of the payout that we want to pay for this specific beneficiary. This is
   // only used when we want to split the payout between multiple beneficiaries
-  const beneficiaryFactor = percentageOfPayout ? Number(percentageOfPayout) / 100 : 1;
+  const beneficiaryFactor = usingPointingSystem
+    ? percentageOfPayout
+      ? Number(percentageOfPayout) / totalPoints
+      : 1
+    : percentageOfPayout
+    ? Number(percentageOfPayout) / 100
+    : 1;
```
[the remark this answers is not in the document]
Comment (shayzluf) [reply packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/usePayoutAllocation.ts:64]: this suggestion looks better

File: packages/web/src/pages/VaultEditor/vaultEditorService.ts [hunk 4]
```diff
@@ -140,33 +145,62 @@ export async function sendEditionToGovApproval(editSessionId: string): Promise<I
...
+            { ...vaultContractInfo, functionName: "decimals" }, // tokenDecimals
+            { ...claimsManagerContractInfo, functionName: "getArbitrator" }, // arbitrator
+            { ...claimsManagerContractInfo, functionName: "arbitratorCanChangeBounty" }, // arbitratorCanChangeBounty
+            { ...claimsManagerContractInfo, functionName: "arbitratorCanChangeBeneficiary" }, // arbitratorCanChangeBeneficiary
+            { ...claimsManagerContractInfo, functionName: "arbitratorCanSubmitClaims" }, // arbitratorCanSubmitClaims
+            { ...claimsManagerContractInfo, functionName: "isTokenLockRevocable" }, // isTokenLockRevocable
+          ] as any),
   }) as Promise<any[]>;
 
   const promisesData = await Promise.all([descriptionHashPromise, allContractCallsPromises]);
```
Comment (shayzluf) [inline packages/web/src/pages/VaultEditor/vaultEditorService.ts:206]: promise all gave us issues. maybe worth to fix it

File: packages/web/src/pages/VaultEditor/vaultEditorService.ts [hunk 5]
```diff
@@ -140,33 +145,62 @@ export async function sendEditionToGovApproval(editSessionId: string): Promise<I
...
+  if (!vault) return undefined;
+  if (vault.version === "v1") return undefined;
+  if (vault.version === "v3" && !vault?.claimsManager) return undefined;
+
   const vaultContractInfo = {
-    address: vaultAddress as `0x${string}`,
-    abi: HATSVaultV2_abi,
-    chainId,
+    address: vault.id as `0x${string}`,
+    abi: vault.version === "v2" ? HATSVaultV2_abi : HATSVaultV3_abi,
```
Comment (shayzluf) [inline packages/web/src/pages/VaultEditor/vaultEditorService.ts:155]: better to be explicit here 
```suggestion
    if (vault.version === "v2") {
  abi = HATSVaultV2_abi;
} else if (vault.version === "v3") {
  abi = HATSVaultV3_abi;
} else {
  // Handle cases where vault.version is neither "v2" nor "v3"
  console.error("Unsupported vault version:", vault.version);
  // You might throw an error or handle it differently based on your needs
}
```

File: packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/usePayoutAllocation.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (fonstack) [reply packages/web/src/pages/CommitteeTools/PayoutsTool/components/PayoutAllocation/usePayoutAllocation.ts:64]: is not doing exactly the same thing...

but I improved the code to:

```
const beneficiaryFactor = percentageOfPayout ? Number(percentageOfPayout) / (usingPointingSystem ? totalPoints : 100) : 1;
```

File: packages/web/src/pages/VaultEditor/vaultEditorService.ts (hunk 4 above)
Comment (fonstack) [reply packages/web/src/pages/VaultEditor/vaultEditorService.ts:206]: in this case makes sense, because if someone fail is because something is wrong...

**Review by shayzluf — APPROVED** [review]

**Pull request merged**